earthfile2llb/withdockerrun.go

```diff
@@ -364,22 +361,12 @@ func (w *withDockerRunTar) solveImage(ctx context.Context, mts *states.MultiTarg
 		if err != nil {
 			return pllb.State{}, errors.Wrapf(err, "build target %s for docker load", opName)
 		}
-		dockerImageID, err := dockertar.GetID(outFile)
-		if err != nil {
-			return pllb.State{}, errors.Wrap(err, "inspect docker tar after build")
-		}
-		// Use the docker image ID + dockerTag as sessionID. This will cause
-		// buildkit to use cache when these are the same as before (eg a docker image
-		// that is identical as before).
-		sessionIDKey := fmt.Sprintf("%s-%s", dockerTag, dockerImageID)
-		sha256SessionIDKey := sha256.Sum256([]byte(sessionIDKey))
-		sessionID := hex.EncodeToString(sha256SessionIDKey[:])
 
 		tarContext := pllb.Local(
 			string(solveID),
-			llb.SessionID(sessionID),
+			llb.SessionID(w.c.opt.GwClient.BuildOpts().SessionID),
```

Comment:
I would not change this part of the code. Note that this is only used for tar-based WITH DOCKER loading, which is no longer used anyway.

Reply:
I spent some time digging in here, and added to the comment explaining what's going on. When I tried removing the call to `llb.SessionID`, the image was nolonger cached -- so I definitely agree with leaving it as-is.

(thanks for pointing out that `VERSION --no-use-registry-for-with-docker 0.6` was needed to test this)